### PR TITLE
Issue #175: Address duplicate code issues from codeclimate

### PR DESCRIPTION
**Pull request by joshua-rutherford**
This removes the duplicate code as reported by code climate.  This does not close the associated issue as there are many more issues with code climate.

### CODE REVIEW

**Comment by joshua-rutherford** [comment]
For what it's worth, this is the code climate improvement for this: https://codeclimate.com/github/DecipherNow/gm-fabric-go/compare/issue-175#issues

**Comment by dougfort** [comment]
CirdleCI is failing on some kind of token error on the oauth unit tests. I suspect it is an intermittent failure that has nothing  to do with these changes.
```
   	github.com/deciphernow/gm-fabric-go/middleware	[no test files]
{"time":"2018-02-21T05:03:11Z","level":"error","error":"Error validating token algorithm: Expected RS256 signing method but token specified HS256","message":"Error validating token algorithm"}
{"time":"2018-02-21T05:03:11Z","level":"error","error":"Error validating token algorithm: Expected RS256 signing method but token specified HS256","message":"Failed to validate token"}
--- FAIL: TestWithToken (0.00s)
	pkg_test.go:66: tokens did not match
FAIL
FAIL	github.com/deciphernow/gm-fabric-go/oauth	0.008s
```

File: oauth/validate.go [hunk 1]
```diff
@@ -153,3 +142,15 @@ func authorize(ctx context.Context, options ValidationOptions, token string) (*P
 		return p, nil
 	}
 }
+
```
Comment (dougfort) [inline oauth/validate.go]: This is failing in the unit test, but I think the code here is OK.

Comment (joshua-rutherford) [reply oauth/validate.go]: Nope, that's a bug.  I'll fix.

Comment (joshua-rutherford) [reply oauth/validate.go]: @dougfort, fixed.  Can you review again?

**Pull request merged**